x/launch/client/cli/tx_request_add_validator.go

```diff
@@ -50,14 +51,29 @@ func CmdRequestAddValidator() *cobra.Command {
 				valAddr = fromAddr
 			}
 
+			valPeerTunnel, _ := cmd.Flags().GetString(flagValidatorPeerTunnel)
+			peer := new(types.Peer)
+			if valPeerTunnel != "" {
+				peer.Connection = &types.Peer_HttpTunnel{
+					HttpTunnel: &types.Peer_HTTPTunnel{
+						Name:    valPeerTunnel,
+						Address: args[4],
+					},
+				}
+			} else {
+				peer.Connection = &types.Peer_TcpAddress{
+					TcpAddress: args[4],
+				}
```

Comment:
I would suggest creating methods in `types/peer.go` to instanciate each type of peer like for other `oneof`